### PR TITLE
Add advancement generator

### PR DESCRIPTION
**Pull request by misode**

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -351,6 +370,45 @@
...
+    "trigger.killed_by_crossbow": "Killed By Crossbow",
+    "trigger.levitation": "Levitation",
+    "trigger.location": "Location",
+    "trigger.nether_travel": "Nether Travel",
+    "trigger.placed_block": "Placed Block",
+    "trigger.player_hurt_entity": "Player Hurt Entity",
+    "trigger.player_killed_entity": "Player Killed Entity",
+    "trigger.recipe_unlocked": "Recipe Unlocked",
+    "trigger.safely_harvest_honey": "Safely Harvest Honey",
+    "trigger.shot_crossbow": "Show Crossbow",
```
Comment (Jerozgen) [inline locales/en.json]: «Shot»

Comment (misode) [reply locales/en.json]: Yep, I had also noticed that 😛

Comment (misode) [reply locales/en.json]: Yep, I had also noticed that 😛

**Comment by misode** [comment]
This branch is missing the `requirements` property, because I'm not sure how to structure that in a compact way. Worst case it would just be a simple text field.

**Comment by misode** [comment]
I also noticed that this breaks the `apply_bonus` function, but since there is a refactor planned I think I can live with that.

**Pull request merged**